File: core/src/net/brenig/pixelescape/screen/PixelScreen.java
package net.brenig.pixelescape.screen;

import com.badlogic.gdx.Screen;

import net.brenig.pixelescape.PixelEscape;

/**
 * Abstract GameScreen providing shared functionality for all Screens of PixelEscape
 */
public abstract class PixelScreen implements Screen {
	public final PixelEscape game;

	public PixelScreen(PixelEscape game) {
		this.game = game;
	}

	public void updateMusic(boolean play) {
		if(play) {
			game.gameMusic.play(true);
		}
	}
}

File: core/src/net/brenig/pixelescape/screen/PixelScreen.kt
package net.brenig.pixelescape.screen

import com.badlogic.gdx.Screen
import de.golfgl.gdxgamesvcs.IGameServiceListener
import net.brenig.pixelescape.PixelEscape
import net.brenig.pixelescape.lib.LogLevel
import net.brenig.pixelescape.lib.log

/**
 * Abstract GameScreen providing shared functionality for all Screens of PixelEscape
 */
abstract class PixelScreen(val game: PixelEscape) : Screen, IGameServiceListener {

    open fun updateMusic(play: Boolean) {
        if (play) {
            game.gameMusic.play(true)
        }
    }

    override fun gsShowErrorToUser(et: IGameServiceListener.GsErrorType?, msg: String?, t: Throwable?) {
        log(LogLevel.WARNING, "GS", "Error: ${et.toString()} - $msg")
    }

    override fun gsOnSessionInactive() {
    }

    override fun gsOnSessionActive() {
    }
}
